packages/jupyter-ai/jupyter_ai/handlers.py

```diff
@@ -121,14 +121,17 @@ def get_chat_user(self) -> ChatUser:
 
         if collaborative:
             names = self.current_user.name.split(" ", maxsplit=2)
-            initials = "".join(
+            # set in case IdentityProvider doesn't return initials, e.g.
+            # JupyterHub (#302)
+            default_initials = "".join(
                 [(name.capitalize()[0] if len(name) > 0 else "") for name in names]
             )
+            current_user_initials = getattr(self.current_user, "initials", None)
```

Comment:
Thanks for catching my error earlier! I've been very busy as of late. 

One last optimization: we only need to define `names` and `default_initials` in the case that the initials are empty. So we can instead write

```py
initials = getattr(self.current_user, "initials", None)
if not initials:
    # compute default initials in case IdentityProvider doesn't
    # return initials, e.g. JupyterHub (#302)
    names = self.current_user.name.split(" ", maxsplit=2)
    initials = "".join(
        [(name.capitalize()[0] if len(name) > 0 else "") for name in names]
    )
```

to avoid defining these in both cases.

And then below, we can define the dict as:

```py
chat_user_kwargs = {
    **asdict(self.current_user),
    "initials": initials,
}
```